core/blockchain.go

```diff
@@ -678,9 +686,23 @@ func (self *BlockChain) LoadLastState(dryrun bool) error {
 	glog.V(logger.Info).Infof("Last block: #%d [%x…] TD=%v", self.currentBlock.Number(), self.currentBlock.Hash().Bytes()[:4], blockTd)
 	glog.V(logger.Info).Infof("Fast block: #%d [%x…] TD=%v", self.currentFastBlock.Number(), self.currentFastBlock.Hash().Bytes()[:4], fastTd)
 
+	self.mu.Unlock()
 	return nil
 }
 
+// PurgeAbove works like SetHead, but instead of rm'ing head <-> bc.currentBlock,
+// it removes all stored blockchain data n -> *anyexistingblockdata*, where
+// scanHeight is how far beyond n should be checked for existing data, ie n+scanHeight recursively
+func (bc *BlockChain) PurgeAbove(n uint64, scanHeight uint64) {
```

Comment:
I don't like the idea of `scanHeight`. I see, that's because we don't know when to end the search, so we're checking in such chunks. But still, we can miss something (extremely rare situation of gap of >2048 blocks). We don't have such methods in database API, but ideally we should iterate the KV store (iterator for prefix + seek) and remove everything. We may switch to this later, after improving database interface. 
And for now, I would prefer to remove parameter, and hardcode 2048 in function body, not in invocation.